### ISSUE TITLE
Post Processing fails: [1c1d327] Problem(s) during processing, failed for the following files/folders: 

### ISSUE THREAD

**Issue by mbrund** [pymedusa/Medusa#10567]
**<Bug Detail>s**
I have Medusa 0.5.1 running on Windows 10.  I have the Post Processing directory setup to pull the SABnzbd downloaded file from E:\Downloads\TV_Completed and my Post directory set to E:\Downloads\TV.  It will fail every time if I don't create the directory C:\Downloads\TV   

To produce the issue; I go to Medusa and do a Forced Download on a saved TV Show.  After the show has downloaded, I go to 'Shows' pull-down, then select Manual Post-Processing; folder = E:\Downloads\TV_Completed, Process = Move, Delete Files = Yes, select PROCESS.  And the failure occurs....

If I build a C:\Downloads\TV directory, it will build the TV Shows main directory name on C:\Downloads\TV\{tv show name} and the directory is empty, but then it will build the same directory on E:\Downloads\TV\{tv show name} and then build out all the episode files on E:\Downloads\TV\{tv show name}.  I have tried it configured with MOVE and COPY and they both fail the same way if C:\Downloads\TV is not present.  

Some additional Background, I started out with SickBeard more than 6 years ago and upgraded to Medusa and running it on the same system all these years. I restored an OS backup a week ago that had Medusa configed back on the system due to HD crash.  I installed a new External USB Hard drive that I use only fo SABnzbd and Medusa file storage that then gets moved to a large NAS after everything is downloaded and renamed.  The original config had the pre and post TV directories configed as C:\Downloads\TV_Completed and C:\Downloads\TV.  I re-configured everything after restoring and changed the GUI config on SABnzbd and Medusa to point to the new directories.   

Its been years since I touched code (retired), but it seems the issue is in post_processor.py, lines 1179-1192
**<end bug details>**

**<post-processor.py code snippet>**
        # if the show directory doesn't exist then make it if desired
        if not os.path.isdir(ep_obj.series._location) and app.CREATE_MISSING_SHOW_DIRS:
            self.log(u"Show directory doesn't exist, creating it", logger.DEBUG)
            try:
                os.mkdir(ep_obj.series._location)
                helpers.chmod_as_parent(ep_obj.series._location)

                # do the library update for synoindex
                notifiers.synoindex_notifier.addFolder(ep_obj.series._location)
            except (OSError, IOError) as error:
                raise EpisodePostProcessingFailedException(u'Unable to create the show directory: {location}. '
                                                           u'Error: {msg}'.format(location=ep_obj.series._location,
                                                                                  msg=error))
**<end code snippet>**

**Medusa Info**
OS: Windows-10-10.0.18362-SP0	
Branch: master
Commit: 1c1d3279ea1c3bb891b8581671e21543aea00c8c
Version: 0.5.1
Database: 44.17
Python Version: 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 22:22:05) [MSC v.1916 64 bit (AMD64)]
Database: 44.17
SSL Version: OpenSSL 1.1.0j 20 Nov 2018
Locale: en_US.cp1252
**<end Medusa Info>**

**Debug logs***
<details>

```
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Missed file: E:\Downloads\TV_Completed\Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS\Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS.mkv: Processing failed: Unable to create the show directory: C:\Downloads\TV\Iron Resurrection. Error: [WinError 3] The system cannot find the path specified: 'C:\\Downloads\\TV\\Iron Resurrection'
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Problem(s) during processing, failed for the following files/folders: 
2022-05-11 01:32:27 INFO	Thread-5 :: [1c1d327] Not deleting folder E:\Downloads\TV_Completed\Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS found the following files: ['Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS.mkv']
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Processing failed for E:\Downloads\TV_Completed\Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS\Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS.mkv: Unable to create the show directory: C:\Downloads\TV\Iron Resurrection. Error: [WinError 3] The system cannot find the path specified: 'C:\\Downloads\\TV\\Iron Resurrection'
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Show directory doesn't exist, creating it
2022-05-11 01:32:27 WARNING	Thread-5 :: [1c1d327] Unable to determine the free space on your OS.
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Trying to determine free space on destination drive
2022-05-11 01:32:27 INFO	Thread-5 :: [1c1d327] This download is marked a priority download so I'm going to replace an existing file if I find one
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] There is no existing file so there's no worries about replacing it
2022-05-11 01:32:27 INFO	Thread-5 :: [1c1d327] New size: 1.23 GB
2022-05-11 01:32:27 INFO	Thread-5 :: [1c1d327] New file: E:\Downloads\TV_Completed\Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS\Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS.mkv
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] This episode is a priority download: True
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Proper: False
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] New quality: 1080p WEB-DL
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Current quality: 1080p WEB-DL
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] NZB: False
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Info hash: None
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Manually snatched: True
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Snatch in history: True
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Quality of the episode we are processing: 1080p WEB-DL
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] This episode was found in history as SNATCHED.
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] The episode file has a quality in it, using that: 1080p WEB-DL
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Retrieving episode object for Iron Resurrection S04E04
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Show information parsed from file name
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Parsed Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS.mkv into title: Iron Resurrection, season: 4, episode: [4], episode_title: Nova Overload, screen_size: 1080p, source: Web, video_codec: H.264, video_encoder: x264, release_group: ROBOTS, container: mkv, mimetype: video/x-matroska, type: episode, parsing_time: 0.057965993881225586, absolute_episode: [], quality: 1080p WEB-DL, total_time: 0.05896592140197754
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Matched release Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS.mkv to a series in your database: Iron Resurrection
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Found numbering S04E04 from parser for show Iron Resurrection S04E04
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Analyzing name: Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS.mkv
2022-05-11 01:32:27 INFO	Thread-5 :: [1c1d327] Processing E:\Downloads\TV_Completed\Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS\Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS.mkv
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Post-processing video files: ['Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS.mkv']
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Post-processing files: ['Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS.mkv']
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] Processing folder: E:\Downloads\TV_Completed\Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS
2022-05-11 01:32:27 DEBUG	Thread-5 :: [1c1d327] No processable items found in folder: E:\Downloads\TV_Completed
2022-05-11 01:32:27 INFO	Thread-5 :: [1c1d327] Processing path: E:\Downloads\TV_Completed
2022-05-11 01:26:54 DEBUG	SNATCHQUEUE-SNATCH-68500 :: [1c1d327] Transaction with 1 queries executed
2022-05-11 01:26:54 DEBUG	SNATCHQUEUE-SNATCH-68500 :: [1c1d327] 68500: Formatting pattern: %SN - %Sx04 - %EN - %QN -> Iron Resurrection - 4x04 - Nova Overload - 1080p WEB-DL
2022-05-11 01:26:54 DEBUG	SNATCHQUEUE-SNATCH-68500 :: [1c1d327] Found codec for Iron Resurrection Nova Overload
2022-05-11 01:26:54 DEBUG	SNATCHQUEUE-SNATCH-68500 :: [1c1d327] Checking if we should stop searching for a better quality for for episode Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS
2022-05-11 01:26:54 DEBUG	SNATCHQUEUE-SNATCH-68500 :: [1c1d327] Result text from SAB: {"status": true, "nzo_ids": ["SABnzbd_nzo_n8we7jyv"]}
2022-05-11 01:26:54 DEBUG	SNATCHQUEUE-SNATCH-68500 :: [1c1d327] User-Agent: Medusa/0.5.1 (Windows; 10; 3a7b3c12-d0f3-11ec-8c31-7cc2c64244d3)
2022-05-11 01:26:54 DEBUG	SNATCHQUEUE-SNATCH-68500 :: [1c1d327] GET URL: https://127.0.0.1:9090/sabnzbd/api?output=json&ma_username=&ma_password=**********&apikey=**********&cat=tv&name=https%3A%2F%2Fnzbfinder.ws%2Fgetnzb%3Fid%3Dbf790231-6260-4ad0-aba9-04df4dc4e4a5.nzb%26r%3D**********&mode=addurl [Status: 200]
2022-05-11 01:26:54 INFO	SNATCHQUEUE-SNATCH-68500 :: [1c1d327] Sending NZB to SABnzbd
2022-05-11 01:26:54 INFO	SNATCHQUEUE-SNATCH-68500 :: [1c1d327] Downloading Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS with size: 1.40 GB from NZB Finder, through a SearchType.MANUAL_SEARCH search
2022-05-11 01:26:54 INFO	SNATCHQUEUE-SNATCH-68500 :: [1c1d327] Beginning to snatch release: Iron.Resurrection.S04E04.Nova.Overload.1080p.WEB.x264-ROBOTS
```
</details>

**<end debug logs>**

**Comment by BenjV** [comment]
Most likely a permission problem.
The user which is running medusa does not have permission to create directories there.

**Comment by mbrund** [comment]
The Medusa service is using Local System account, it has more authority than say an Administrator, so how would that be possible?

If I mkdir C:\Downloads\TV, and say I download TV Show NCIS - S14E09, Medusa will mkdir C:\Downloads\TV\NCSI and that's all, it will then mkdir E:\Downloads\TV\NCIS\Season 14 and place all the files there.   It seems to me that the Python script is using the wrong directory structure in the first part of the script somewhere.  It's always defaulting to C: first even though I have Post-Processing DIR set in Post-Processing->Episode Naming->Name Pattern set to: E:\Downloads\TV\%SN/Season %S/%SN - S%0SE%0E - %EN, and then if it can build the C: Show Name dir it will proceed to build the E: show name dir, epsiode dir, and then move the files from E:\DownloadTV_Completed there and rename them as requested.

**Comment by mbrund** [comment]
I had a little time yesterday to look at it some more, since I am not  familiar with Medusa or Python I am not able to go further right now.  I found if I hard code line 1180 from:
if not os.path.isdir(ep_obj.series._location) and app.CREATE_MISSING_SHOW_DIRS:

To: 

if not os.path.isdir('E:\\Downloads\\TV') and app.CREATE_MISSING_SHOW_DIRS:

The script no longer has the error about not being able to build the directory on C:, and builds the directory on E: and moves the files there as it should.   So something is amiss with the string: ep_obj.series._location).  IT would seem it should be using something like self.folder_path

**Comment by BenjV** [comment]
That is not correct.
You should have used os.path.isdir('E:\\Downloads\\TV'), because in python you need to escape a backslash with a second backslash.
Python would translate this to os.path.isdir('E:DownloadsTV') and that certainly does not exists and so it create the folder.

The conclusion must be that the original code (e.g. f not os.path.isdir(ep_obj.series._location) ) concluded that the folder is already there and not needed to create.
Maybe the drive letter is somehow lost, because on other os-en there is no such thing as a drive letter and Medusa is developed on Linux.

You can check this by activating the Debug log and look at the debug messages.
Put this code in a line before the if statement to see what is happening
`self.log(f"Needed folder: {ep_obj.series._location}", logger.DEBUG)`

Be careful, python is position sensitive so be sure to start this code on the same column as the if statement.

**Comment by mbrund** [comment]
Sorry, I mis-typed, it was escaped with double backslash.  I tried using forward slashes in the Medusa configuration file for the target and it didn't matter.  I will add the logging statement when I get home this eve and see what it places in the debug log.  I thought I did that, and it stated C:\yadayadayada, but I will double check and respond back.

**Comment by BenjV** [comment]
Ha ha, in the markup language for the postings here you also need to escape the backslash, so to show here a double backslash one needs to enter 4 backslashes.

**Comment by mbrund** [comment]
Here is the results:  
**<DEBUG LOG>**
022-05-12 20:12:40 DEBUG	Thread :: [1c1d327] Needed folder: C:\Downloads\TV\Chrisley Knows Best
**<DEBUG LOG END>**

Its clearly the wrong drive, as E:\\\\Downloads\\\\TV is configured in Medusa Post-Processing->Episode Naming->Name Pattern

**Comment by p0psicles** [comment]
To start of medusa version 0.5.1 is already an ancient one. We're currently on https://github.com/pymedusa/Medusa/releases/tag/v0.5.29. So i'd recommend to at least upgrade to the latest.

Then i'd like to get a better view of your postprocessing process. Like what's configured for the show's final location.
Maybe create some screenshots of the config - postprocessing page. And of the show page. And maybe others if they could be relevant.

**Comment by mbrund** [comment]
OK, I ran the check for updates off the application tab, and it indicates I am up to date.  I checked the change log and it shows latest is  0.5.29.  So I will need to do a pull and install that version in the PM, need to get some sleep now.  After I install, I will check to see if it still has the same issue.  I kind of expect it to, as I dont think any changes have been done to that section of code to make it start using something like DEFAULT_CLIENT_PATH.  But will update the thread afterwards,.

**Comment by mbrund** [comment]
Installled 0.5.29 and same issue. I even installed Medusa on the E: drive this time with all the databases, TV/Movies Downloads directories, etc.  And it still insist on building the C:\Downloads\TV\{showname} directory....

**<MANUAL POST-PROCESSING RESULTS>**
...
Processing path: E:\Downloads\TV_Completed
No processable items found in folder: E:\Downloads\TV_Completed
No resource_name passed, using path [E:\Downloads\TV_Completed\Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE] to process as a source folder
Processing folder: E:\Downloads\TV_Completed\Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE
Post-processing files: ['Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE.mkv']
Post-processing video files: ['Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE.mkv']
Processing failed for E:\Downloads\TV_Completed\Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE\Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE.mkv: Unable to create the show directory: C:\Downloads\TV\Chrisley Knows Best. Error: [WinError 3] The system cannot find the path specified: 'C:\\Downloads\\TV\\Chrisley Knows Best'
Failed download detected: (resource: Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE.mkv, dir: E:\Downloads\TV_Completed\Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE)
Parsed info: title: Chrisley Knows Best, season: 8, episode: [13], screen_size: 1080p, source: Web, video_codec: H.264, release_group: BAE, type: episode, parsing_time: 0.0519709587097168, absolute_episode: [], quality: 1080p WEB-DL, total_time: 0.05396866798400879
Detected as single/multi episode: Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE
Created segment of episodes [13] from release: Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE
Deleted folder: E:\Downloads\TV_Completed\Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE
Failed Download Processing succeeded: None, E:\Downloads\TV_Completed\Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE
Problem(s) during processing, failed for the following files/folders: 
Missed file: E:\Downloads\TV_Completed\Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE\Chrisley.Knows.Best.S08E13.1080p.WEB.h264-BAE.mkv: Processing failed: Unable to create the show directory: C:\Downloads\TV\Chrisley Knows Best. Error: [WinError 3] The system cannot find the path specified: 'C:\\Downloads\\TV\\Chrisley Knows Best'
...
**<END MANUAL POST-PROCESSING RESULTS>**

**Comment by p0psicles** [comment]
What's the root dit for that show? How is the show configured? Screenshots?

**Comment by mbrund** [comment]
Thanks for making me look at the Show Database file, that is where the issue resided.  When I built my SickRage many years ago everything was on C:.  So as I have updated and converted over the years I have just ported over the database with each update by leaving the db files in place and letting the installs use them.  Well when I lost the hard drive a couple weeks ago and installed a new C: drive, I also installed a new external USBN 3.0 hard drive as E:.  When I restored my OS and Apps Backup, I went in and changed the config file, but did not think that the database had hard drive assignments, thought it was just directory related.  

So when I just went to do a screen grab of the show for you, I saw it stated there is BIG OLD LETTERS that the shows location is C:\Downloads\TV\{show name}. So I changed it, ran the post processing and all is good.... (⌒_⌒;)
